x-pack/legacy/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_list/action_clone.ts

```diff
@@ -0,0 +1,249 @@
+              defaultValue: 0.05,
+              optional: true,
+            },
+            feature_influence_threshold: {
+              optional: true,
+            },
+            method: {
+              optional: true,
+            },
+            n_neighbors: {
+              optional: true,
+            },
+          },
+        }
+      : {}),
+    ...(isRegressionAnalysis(config.analysis)
+      ? {
+          regression: {
+            dependent_variable: {
+              optional: false,
+              formKey: 'dependentVariable',
+            },
+            training_percent: {
+              optional: true,
+              formKey: 'trainingPercent',
+            },
+            eta: {
+              optional: true,
+            },
+            feature_bag_fraction: {
+              optional: true,
+            },
+            maximum_number_trees: {
+              optional: true,
+            },
+            gamma: {
+              optional: true,
+            },
+            lambda: {
+              optional: true,
+            },
+            prediction_field_name: {
+              optional: true,
+              defaultValue: `${config.analysis!.regression!.dependent_variable}_prediction`,
+            },
+            num_top_feature_importance_values: {
+              optional: true,
+            },
+            randomize_seed: {
+              optional: true,
+              // By default it is randomly generated
+            },
+          },
+        }
+      : {}),
+  },
+  analyzed_fields: {
+    excludes: {
+      optional: true,
+      formKey: 'excludes',
+      defaultValue: [],
+    },
+    includes: {
+      optional: true,
+      defaultValue: [],
+    },
+  },
+  source: {
+    index: {
+      formKey: 'sourceIndex',
+      optional: false,
+    },
+    query: {
+      optional: true,
+      defaultValue: {
+        match_all: {},
+      },
+    },
+    _source: {
+      optional: true,
+    },
+  },
+  dest: {
+    index: {
+      optional: false,
+    },
+    results_field: {
+      optional: true,
+      defaultValue: 'ml',
+    },
+  },
+  model_memory_limit: {
+    optional: true,
+    formKey: 'modelMemoryLimit',
+  },
+});
+
+/**
+ * Detects if analytics job configuration were created with
+ * the advanced editor and not supported by the regular form.
+ */
+export function isAdvancedConfig(config: any, meta?: AnalyticsJobMetaData): boolean;
+export function isAdvancedConfig(
+  config: DataFrameAnalyticsConfig,
+  meta: AnalyticsJobMetaData = getAnalyticsJobMeta(config)
+): boolean {
+  for (const configKey in config) {
+    if (config.hasOwnProperty(configKey)) {
+      const fieldConfig = config[configKey as keyof typeof config];
+      const fieldMeta = meta[configKey as keyof typeof meta];
+
+      if (fieldMeta) {
+        if (isPropDefinition(fieldMeta)) {
+          const isAdvancedSetting =
+            !fieldMeta.formKey &&
+            fieldMeta.defaultValue !== undefined &&
```

Comment:
Is this check for a `defaultValue` correct? For instance, I created a classification job, where I had set `maximum_number_trees` in the advanced editor. But when I clone the job, it opens into the form editor, and my `maximum_number_trees` setting is lost.

The config I had in the advanced editor for the original job was:

```
{
  "description": "Classification clone with 'bank-marketing' dataset",
  "source": {
    "index": "bank-marketing"
  },
  "dest": {
    "index": "bank_classification4"
  },
  "analyzed_fields": {
    "excludes": []
  },
  "analysis": {
    "classification": {
      "dependent_variable": "y",
      "training_percent": 71,
      "maximum_number_trees": 1500
    }
  },
  "model_memory_limit": "400mb"
}
```

Reply:
Fixed in https://github.com/elastic/kibana/pull/59791/commits/ba2cabd1aff904c1b8e033657d7960bfb4abe94d